tests/random_test.py

```diff
@@ -1107,6 +1107,20 @@ class A: pass
     with self.assertRaisesRegex(TypeError, 'unrecognized type .* PRNG'):
       jax.random.key(42, impl=A())
 
+  def test_keyarray_custom_vjp(self):
+    # Regression test for https://github.com/google/jax/issues/18442
+    @jax.custom_vjp
+    def f(_, state):
+      return state
+    def _f_fwd(_, state):
+      return state, None
+    def _f_bwd(_, state_bar):
+      return state_bar
```

Comment:
What does the key part of `state_bar` look like here? A valid key array backed by 0 values?

Reply:
Yes, it's a valid key array backed by zeros.

Reply:
Seems unsafe, no? We don't want such a key to be used.

Tagging @mattjj because we've pondered this in the past.

This may also relate to `jnp.zeros_like` (which implements a stopgap at the moment, arguably), but not necessarily.

Reply:
If we get key reuse working, it could be a consumed key 😁

Reply:
OK, changed this to a special key tangent type, so it can't be accidentally used as a key. It will fail very ungracefully if it's used for anything, but we can figure that out later.